source/ConditionsStore.h

```diff
@@ -0,0 +1,152 @@
+	// for displaying some data based on primary conditions.
+	class PrimariesIterator: public std::iterator<
+		std::input_iterator_tag,                      // iterator_category
+		std::pair<const std::string, int64_t>,        // iterator: value_type
+		std::ptrdiff_t,                               // iterator: difference_type
+		const std::pair<const std::string, int64_t>*, // iterator: pointer
+		std::pair<const std::string, int64_t>>        // iterator: reference
+	{
+		using CondMapItType = std::map<std::string, ConditionEntry>::const_iterator;
+	
+	public:
+		PrimariesIterator(CondMapItType it, CondMapItType endIt);
+		
+		// Default input_iterator operations.
+		std::pair<std::string, int64_t> operator*() const;
+		const std::pair<std::string, int64_t>* operator->();
+		PrimariesIterator& operator++();
+		PrimariesIterator operator++(int);
+		bool operator== (const PrimariesIterator& rhs) const;
+		bool operator!= (const PrimariesIterator& rhs) const;
+	
+	
+	private:
+		// Helper function to ensure that the primary-conditions iterator points
+		// to a primary (value) condition or to the end-iterator value.
+		void MoveToValueCondition();
+	
+	
+	private:
+		// The operator->() requires a return value that is a pointer, but in this
+		// case there is no original pair-object to point to, so we generate a
+		// virtual object on the fly while iterating.
+		std::pair<std::string, int64_t> itVal;
+		CondMapItType condMapIt;
+		CondMapItType condMapEnd;
+	};
+
+
+
+public:
+	// Constructors to initialize this class.
+	ConditionsStore() = default;
+	ConditionsStore(std::initializer_list<std::pair<std::string, int64_t>> initialConditions);
+	ConditionsStore(const std::map<std::string, int64_t> &initialConditions);
+
+	// Retrieve a "condition" flag from this store (directly or from the
+	// connected provider).
+	int64_t GetCondition(const std::string &name) const;
+	bool HasCondition(const std::string &name) const;
+	// Add a value to a condition, set a value for a condition or erase a
+	// condition completely. Returns true on success, false on failure.
+	bool AddCondition(const std::string &name, int64_t value);
+	bool SetCondition(const std::string &name, int64_t value);
+	bool EraseCondition(const std::string &name);
+	
+	// Direct (read-only) access to the stored primary conditions.
+	PrimariesIterator PrimariesBegin() const;
+	PrimariesIterator PrimariesEnd() const;
+	PrimariesIterator PrimariesLowerBound(const std::string &key) const;
+	
+	// Set providers for derived conditions based on prefix and name.
+	void SetProviderPrefixed(const std::string &prefix, DerivedProvider conditionsProvider);
+	void SetProviderNamed(const std::string &name, DerivedProvider conditionProvider);
+	
+	// Helper to completely remove all data and linked condition-providers from the store.
+	void Clear();
+
+
+private:
+	// Retrieve a condition entry based on a condition name, doCreate indicate if the entry
+	// should be created if it wasn't there yet.
```

Comment:
Wahts `doCreate` that this comment alludes to?

Reply:
An artifact of some old code (where the has-function should never create new entries, while the setFun should). I think I can completely remove this comment now.